OpenAIChatGPTBlazor/Pages/Index.razor.cs

```diff
@@ -62,23 +68,27 @@ private async Task RunSearch()
                 this.StateHasChanged();
                 _chat.Messages.Add(new ChatMessage(ChatRole.User, _next));
                 _next = string.Empty;
-                var res = await OpenAiClient.GetChatCompletionsStreamingAsync(_selectedModel, _chat);
-                await foreach (var choice in res.Value.GetChoicesStreaming())
+                _searchCancellationTokenSource = new CancellationTokenSource();
+                var res = await OpenAiClient.GetChatCompletionsStreamingAsync(_selectedModel, _chat, _searchCancellationTokenSource.Token);
+                await foreach (var choice in res.Value.GetChoicesStreaming(_searchCancellationTokenSource.Token))
                 {
-                    await foreach (var msg in choice.GetMessageStreaming())
+                    await foreach (var msg in choice.GetMessageStreaming(_searchCancellationTokenSource.Token))
                     {
-                        // We collect response as it comes in and display it immediately
                         _stream += msg.Content;
                         this.StateHasChanged();
-                        await JS.InvokeVoidAsync("scrollElementIntoView", _nextArea);
+                        await JS.InvokeVoidAsync("scrollElementToEnd", _mainArea);
                     }
                 }
 
-                _chat.Messages.Add(new ChatMessage(ChatRole.Assistant, _stream));
+                _chat.Messages.Add(new ChatMessage(ChatRole.Assistant, _stream.ToString()));
                 _loading = false;
                 _stream = string.Empty;
                 _warningMessage = string.Empty;
             }
+            catch (TaskCanceledException)
+            {
+                // Currently (2023-06-18) ignoring due to the OpenAiClient implementation not seeming to properly handle the cancellation.
```

Comment:
I adapted it slightly

Reply:
I can't adapt since I can't push to your repo :D anyways I just made it
```
            catch (TaskCanceledException) when (_searchCancellationTokenSource?.Token.IsCancellationRequested == true)
            {
                // Graceful cancellation
            }
```